### ISSUE TITLE
feat: Accessibility Improvements Across Multiple Pages

### ISSUE THREAD

**Issue by seangray-dev** [seangray-dev/formail-app#35]
## Description
Multiple accessibility issues have been identified across various pages of our application. These issues range from improper ARIA attribute usage to insufficient button naming and color contrast problems. Addressing these will improve our compliance with WCAG guidelines and enhance usability for all users, especially those relying on assistive technologies.

### Dashboard Page
- [x] **[aria-*] Attributes and Roles**: The 'Create New Form' button has `[aria-*]` attributes that do not match their roles.
- [x] **Buttons and Links Naming**: Multiple buttons and links lack accessible names, affecting screen reader users.

### Submissions Page
- [x] **Buttons Naming**: Multiple buttons do not have accessible names.

### Analytics Page
- [x] **Buttons Naming**: The 'Submissions By' button lacks an accessible name.
- [x] **Progress Bars**: ARIA `progressbar` elements are missing accessible names.

### Export Page
- [x] **Buttons Naming**: The 'Select a Format to Export' button lacks an accessible name.

### Settings Page
- [x] **Buttons Naming**: Accessibility issues with the naming of checkbox and toggle button in the email notifications and email threads sections.
- [x] **Color Contrast**: The 'Delete All Submissions' and 'Delete Form' buttons have insufficient color contrast between background and foreground.

### How-To Page
- [x] **Heading Order**: The 'Need Help Alert' uses an `<h5>` tag that does not follow a sequentially-descending order.

**Comment by seangray-dev** [comment]
also fixed accessibility issues on pricing and contact pages
